docs/Prodigy_training_data.md

```diff
@@ -145,53 +129,83 @@ A comparison with the previous results (201022 - old definition, 210128 - new de
+| 210221 | bert | naive_bayes | scibert | 0.764 | 0.779 | 0.803 | 0.756 |
 | 201022 | bert | SVM | scibert | 0.776 | 0.879 | 0.904 | 0.855 |
 | 210128 | bert | SVM | scibert | 0.831 | 0.851 | 0.819 | 0.885 |
-| 210218 | bert | SVM | scibert | 0.813| 0.817 | 0.849 | 0.787 |
+| 210221 | bert | SVM | scibert | 0.793 | 0.783 | 0.839 | 0.733 |
 | 201022 | bert | log_reg | scibert | 1.000 | 0.814 | 0.762 | 0.873 |
 | 210128 | bert | log_reg | scibert | 1.000 | 0.775 | 0.758 | 0.793 |
-| 210218 | bert | log_reg | scibert | 0.997| 0.853 | 0.853 | 0.853 |
+| 210221 | bert | log_reg | scibert | 1.000 | 0.794 | 0.819 | 0.770 |
 
 ### Ensemble model
 
-I ran:
+In the previous method we ran the script `ensemble_model.py` to make predictions using an ensemble of models which satisfied precision and recall value thresholds. However in this iteration we improved the process by testing which ensemble of models and which number of models that needs to agree is optimal.
+
+We first make predictions using the test data for each of the 12 models. This is done by running:
+```
+python nutrition_labels/predict_tech.py 
+    --training_data_file data/processed/training_data/210221/training_data.csv
+    --grants_data_file data/raw/wellcome-grants-awarded-2005-2019.csv
+    --models_date 210221
+```
+
+The ensemble parameter experiments were run in `notebooks/Ensemble parameter exploration.ipynb`. In this we plot the precision/recall for each of the 12 models plus every single combination of those 12 models for different values of the number of models that need to agree.
+
+![](figures/model_performance_ensemble_vs_single.png)
+
+The models found with the 'best' recall and precision were:
+
+- Ensemble of 3 models with 2 needing to agree ('bert_log_reg_bert_210221', 'bert_SVM_bert_210221', 'count_naive_bayes_210221')
+- Ensemble of 4 models with 3 needing to agree ('bert_naive_bayes_bert_210221', 'bert_log_reg_bert_210221', 'bert_SVM_bert_210221', 'count_naive_bayes_210221')
+- Ensemble of 4 models with 3 needing to agree ('bert_log_reg_bert_210221', 'bert_SVM_bert_210221', 'bert_naive_bayes_scibert_210221', 'count_naive_bayes_210221')
+- Ensemble of 3 models with 4 needing to agree ('bert_naive_bayes_bert_210221', 'bert_log_reg_bert_210221', 'bert_SVM_bert_210221', 'bert_naive_bayes_scibert_210221', 'count_naive_bayes_210221')
+
+out of these we chose the ensemble with:
+- 3 models ('bert_log_reg_bert_210221', 'bert_SVM_bert_210221', 'count_naive_bayes_210221')
+- 2 need to agree on the grant being a tech grant in order for final prediction to be tech.
+
+we chose this since making predictions using fewer models is preferred in terms of time resource.
+
+The classification report for this ensemble model is:
+
+||precision|recall|f1-score|support|
+|--|--|--|--|--|
+|0|0.78|0.82|0.80|110|
+|1|0.85|0.81|0.83|135|
+|accuracy|||0.82|245|
+|macro avg|0.81|0.82|0.82|245|
+|weighted avg|0.82|0.82|0.82|245|
+
+Finally these details were put into a config file and
```

Comment:
ok this is where the config files comes from. note that it is better for reproducibility to submit configs on all models you report results in, not just the ones you "productionise"